### PR TITLE
Add LiteLoaderBDS egg

### PR DESCRIPTION
**Pull request by Lycol50**
# Description

Creation of LiteLoaderBDS egg support

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [X] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [X] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [X] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?

### CODE REVIEW

File: game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/parkervcp\/yolks:wine_latest": "ghcr.io\/parkervcp\/yolks:wine_latest"
+    },
+    "file_denylist": [],
+    "startup": ".\/lae-ll-launcher",
+    "config": {
+        "files": "{\r\n    \"server.properties\": {\r\n        \"parser\": \"properties\",\r\n        \"find\": {\r\n            \"server-port\": \"{{server.build.default.port}}\",\r\n            \"server-ip\": \"0.0.0.0\"\r\n        }\r\n    }\r\n}",
+        "startup": "{\r\n    \"done\": \"[LiteLoader]\"\r\n}",
+        "logs": "{\r\n    \"custom\": false,\r\n    \"location\": \"logs/latest.log\"\r\n}",
```
Comment (QuintenQVD0) [inline game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json:20]: This not not used anymore

File: game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string",
+            "field_type": "text"
+        },
+        {
+            "name": "Liteloader Version",
+            "description": "Insert Liteloader Version from Github Releases",
+            "env_variable": "LL_VERSION",
+            "default_value": "2.8.1",
```
Comment (QuintenQVD0) [inline game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json:45]: This by default should be latest

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

**Comment by QuintenQVD0** [comment]
Alos where are your commit to adding it to the main readme and sub readme of game eggs a Minecraft and bedrock

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "startup": ".\/lae-ll-launcher",
+    "config": {
+        "files": "{\r\n    \"server.properties\": {\r\n        \"parser\": \"properties\",\r\n        \"find\": {\r\n            \"server-port\": \"{{server.build.default.port}}\",\r\n            \"server-ip\": \"0.0.0.0\"\r\n        }\r\n    }\r\n}",
+        "startup": "{\r\n    \"done\": \"[LiteLoader]\"\r\n}",
+        "logs": "{\r\n    \"custom\": false,\r\n    \"location\": \"logs/latest.log\"\r\n}",
+        "stop": "stop"
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/bash\r\n# Liteloader First-time Install Script\r\n\r\nrm -r \/mnt\/server\r\nmkdir \/mnt\/server\r\ncd \/mnt\/server\r\nwget https:\/\/minecraft.azureedge.net\/bin-win\/bedrock-server-${BDS_VERSION}.zip -O bds.zip\r\nwget https:\/\/github.com\/LiteLDev\/LiteLoaderBDS\/releases\/download\/${LL_VERSION}\/LiteLoader-${LL_VERSION}.zip -O LiteLoader.zip\r\nwget https:\/\/io.ivampiresp.com\/d\/Software\/MCServer\/BDS\/LiteLoader\/lae-ll-launcher\r\nchmod +x lae-ll-launcher\r\nunzip -o bds.zip\r\nunzip -o LiteLoader.zip\r\nrm LiteLoader.zip\r\nrm bds.zip\r\necho \"Finished Downloading Liteloader...\"",
```
[the remark this answers is not in the document]
Comment (Lycol50) [reply game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json]: acknowledged

File: game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            "container": "ghcr.io\/parkervcp\/installers:alpine",
+            "entrypoint": "ash"
+        }
+    },
+    "variables": [
+        {
+            "name": "Bedrock Dedicated Server Version",
+            "description": "Insert BDS Version",
+            "env_variable": "BDS_VERSION",
+            "default_value": "1.19.41.01",
```
[the remark this answers is not in the document]
Comment (Lycol50) [reply game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json]: acknowledged

File: game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json (hunk 2 above)
Comment (Lycol50) [reply game_eggs/minecraft/bedrock/LiteLoader-bedrock/egg-LiteLoader-bedrock.json]: acknowledged

**Comment by QuintenQVD0** [comment]
The downloading of the binary's of LiteLoaderBDS should be done with curl and not wget
and the downloading should be done outside the script to get the version so only 1 line is needed to download the binary. Also please use our github script to get the versions

**Comment by parkervcp** [comment]
> done with curl and not wget

Both are fine as long as they get the files.

**Comment by QuintenQVD0** [comment]
wy did you close it? for the rest it looks fine

**Comment by Lycol50** [comment]
> wy did you close it? for the rest it looks fine

You didn't review the changes, before you comment.
1. The last commits are now just the same with Bedrock Server Egg, so hence it uses the github script
2. wget and curl are just the same (to get files)

I will not re-open this pull request anymore. Thanks

**Comment by QuintenQVD0** [comment]
> > wy did you close it? for the rest it looks fine
> 
> You didn't review the changes, before you comment.
> 
>     1. The last commits are now just the same with Bedrock Server Egg, so hence it uses the github script
> 
>     2. wget and curl are just the same (to get files)
> 
> 
> ~I will not re-open this pull request anymore. Thanks~

Then it looks fine to me. I don't have time ATM but if someone can confirm that it works and can join the server then this will be merged

**Comment by Lycol50** [comment]
> I tested the egg and need to changes:
> 
> * Move to wine staging image
> * set the startup done to `IPv4 supported`

may I know why it need to use the wine staging image, instead of wine latest one?

**Comment by QuintenQVD0** [comment]
> > I tested the egg and need to changes:
> > 
> > * Move to wine staging image
> > * set the startup done to `IPv4 supported`
> 
> may I know why it need to use the wine staging image, instead of wine latest one?

Wine staging receives way more bug fixes, if anything changes to LiteLoaderBDS  code then likly wine staging will handel it beter as it gets way more updates ( wine is a v 7.0 and staging at 7.22)

**Comment by QuintenQVD0** [comment]
also please use the panel export feature as then the exported time changes, we may need this in the future

**Comment by Lycol50** [comment]
okay, i understand. This will be noted in the future.

**Comment by QuintenQVD0** [comment]
LGTM! Thank you!

**Pull request merged**